src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs

```diff
@@ -190,12 +190,21 @@ public async Task<DicomDataset> GetWorkitemBlobAsync(WorkitemInstanceIdentifier
             .ConfigureAwait(false);
     }
 
+    /// <inheritdoc />
+    public async Task<DicomDataset> RetrieveWorkitemAsync(WorkitemInstanceIdentifier workitemInstanceIdentifier, CancellationToken cancellationToken = default)
+    {
+        EnsureArg.IsNotNull(workitemInstanceIdentifier, nameof(workitemInstanceIdentifier));
+
+        return await TryGetWorkitemBlobAsync(workitemInstanceIdentifier, cancellationToken)
+            .ConfigureAwait(false);
+    }
+
     private async Task<DicomDataset> TryGetWorkitemBlobAsync(WorkitemInstanceIdentifier identifier, CancellationToken cancellationToken)
     {
         try
         {
             return await GetWorkitemBlobAsync(identifier, cancellationToken)
-            .ConfigureAwait(false);
+                .ConfigureAwait(false);
```

Comment:
Could you remove logging ids atleast for the method you are adding.

Reply:
Identifier to string logs only PartitionKey, WorkitemKey, and Watermark.